Modules/ThirdParty/GDCM/src/gdcm/Source/DataStructureAndEncodingDefinition/gdcmDataSet.h

```diff
@@ -209,13 +209,9 @@ class GDCM_EXPORT DataSet
 
   // DUMB: this only search within the level of the current DataSet
   bool FindDataElement(const Tag &t) const {
-    const DataElement r(t);
-    //ConstIterator it = DES.find(r);
-    if( DES.find(r) != DES.end() )
-      {
-      return true;
-      }
-    return false;
+    const auto it = GetDataElement(t);
```

Comment:
BTW,  this will highly likely trigger _copy_ of the DataElement (the function returns `const DataElement&`, but this `const auto =` forces copy of the object, IMO). It is not a request to change, the issue is very odd. cc @N-Dekker

Reply:
@issakomi I just had a few days off, sorry for my late response! I'm not very familiar with this code. I would expect that it would search in the DataElementSet for a DataElement with the specified tag. But instead, it locally creates a DataElement with the specified tag, using default values for its VL and VR. It then tries to find this local DataElement in the DataElementSet. So _I think_ it can only find a DataElement with those default values for VL and VR. Is that OK? 

Anyway, I agree with you that a "copy" might be skipped by doing `return (GetDataElement(t) != GetDEEnd())`.

Reply:
I have done some test (Linux, GCC 12).  These _GetDataElement_ and _FindDataElement_ rely on _DES.find_. DES is `std::set<DataElement>`. Interestingly DES.find doesn't seem to use DataElement's [operator==](https://github.com/InsightSoftwareConsortium/ITK/blob/28de86169f65ad8a9800b6cadc535e7f54c65a55/Modules/ThirdParty/GDCM/src/gdcm/Source/DataStructureAndEncodingDefinition/gdcmDataElement.h#L172-L187). The function works (both old and new variants) with GCC 12, but I still don't fully understand ...

Reply:
It seems that comparing for _find_ is something like `!(x < y) && !(y < x)`
without `operator==`, only with `operator<`. The last one uses `Tag`. But what was the issue with GCC 13 is still not clear.

Reply:
@issakomi Thanks for this little research! 👍

Reply:
Interesting that GDCM almost always does something like
```
  if( !subds2.FindDataElement(tps) ) return false;
  const DataElement &de = subds2.GetDataElement( tps );
```
FindDataElement does in fact 1st GetDataElement and 2nd compares returned one with invalid DataElement(Tag(0xffff, 0xffff)). So far 1 _find_ too much (and it is not cheap: "Complexity: logarithmic in the size of the container")
It could be e.g. or something
```
  const DataElement &de = subds2.GetDataElement( tps );
  if (de == DataElement(Tag(0xffff, 0xffff)) return false;
```
There are dozens or hundreds of such double _find_ s in e.g. gdcmImageHelper. Sometimes 3 times, e.g. in [itkGDCMImageIO](https://github.com/InsightSoftwareConsortium/ITK/blob/28de86169f65ad8a9800b6cadc535e7f54c65a55/Modules/IO/GDCM/src/itkGDCMImageIO.cxx#L650-L652)
Just FYI. I will not start PRs for this.